cgp/local_search/gradient_based.py

```diff
@@ -57,20 +57,33 @@ def gradient_based(
 
     f = individual.to_torch()
 
-    if len(list(f.parameters())) > 0:
-        optimizer = optimizer_class(f.parameters(), lr=lr)
+    if isinstance(f, list):
```

Comment:
this stuff makes me flinch every time i see it, but i don't think we have an easier solution. oh well :|